java/src/Ice/src/main/java/com/zeroc/Ice/_ObjectPrxIWrapper.java

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) ZeroC, Inc. All rights reserved.
+
+package com.zeroc.Ice;
+
+/**
+ * @hidden Provides overloads of the proxy factory methods with covariant return types so that
+ *     applications do not need to downcast to the derived proxy type.
+ */
+@SuppressWarnings("unchecked")
+public abstract class _ObjectPrxIWrapper<T extends ObjectPrx> extends _ObjectPrxI {
```

Comment:
This class doesn't wrap anything as far as I can tell. What about:
`ProxyFactoryMethods<T>` or `_ProxyFactoryMethods<T>`?

Reply:
Yes, you're correct. I changed the name (but kept the underscore to signal that it's _kind of_ internal.